strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/AbstractLayoutProvider.java

```diff
@@ -169,7 +169,14 @@ protected boolean isChecksum(String fileName)
                 value = repositoryPath.getRepository().getId();
 
                 break;
-                
+            case EXPIRED:
+                value = false;
+
+                break;
+            case REQUIRES_GROUP_AGGREGATION:
```

Comment:
This seems to me that such attribute is not really good thing for common attributes, because in fact it's only for maven. 
Maybe we should consider new method in `RepositoryProvider` for repository group metadata resolution as was suggested in channel?

Reply:
ok, after revisiting SB-1309 it seems I see what you mean with `firstMatch` logic in `GroupRepositoryProvider` and this attribute, but this attribute special case for real :)
so I suggest to change it with sync event like we have for expired metadata fetch.

Reply:
I'll consider that in the evening

Reply:
I think we don't need to make it sync. Here is the most common case:

1. Request for `maven-metadata.xml`
2. Fetch remote checksum of `maven-metadata.xml`
3. Compare remote checksum with local checksum
4. If checksums don't match, re-fetch `maven-metadata.xml`
5. Storing calculated checksum for re-fetched `maven-metadata.xml` may be done in async, since the  `maven-metadata.xml` will expire after a minute (we can assume async event will be called within this time).

If my assumptions are bad, we will probably be notified about it in Jenkins.

Reply:
yes, but I mean not this :)
I mean that we shouldn't add this extra `REQUIRES_GROUP_AGGREGATION` attribute.
Currently you using it to get ` boolean firstMatch` and then decide that you need to aggregate metadata based on it.
So we should drop `REQUIRES_GROUP_AGGREGATION` and trigger an event just like you have it in `ProxyRepositoryProvider` for expired paths.

Reply:
ok, it can be done this way too.

Done.